### DISCUSSION TITLE
Show/hide column based on filter selection

### DISCUSSION THREAD

**Discussion by thaisonle** [filamentphp/filament#5606] [Ideas]
My use case is a resource table where I'm displaying in-progress and submitted forms and I have a TernaryFilter, based off a `submitted` boolean on the Form model, that shows 'All', 'Only Submitted', or 'Only In-Progress' forms based on empty, true, and false, respectively. 

I would like to show a 'Created At' column when 'In-Progress' is selected, a 'Submitted At' column when 'Submitted' is selected, or both if 'All' is selected.

I'm pretty sure this functionality doesn't currently exist but it'd be awesome if I'm wrong.

Kinda similar to this request #690

**Comment by danharrin** [comment]
You could do this with a SelectFilter and a `query()` callback to decide how to modify the query based on the selected option

**Reply by thaisonle** [reply]
@danharrin I'm not sure that's what I'm looking for, I understand how to customize my query using the callback. I'm not trying to do that, I'm basically trying to change the `columns()` configuration dynamically. I thought I could maybe mess with the ComponentContainer and manually add/remove components but it's protected.

**Reply by danharrin** [reply]
Sorry, I misunderstood you.

Try something like `->hidden(fn ($livewire) => $livewire->getTableFilterState('FILTER_NAME')['value'] === 'value')`. `getTableFilterState()` returns the data from the filter and you can check it
